src/account/account.cairo

```diff
@@ -0,0 +1,218 @@
+// }
+
+const QUERY_VERSION: felt252 = 340282366920938463463374607431768211457;
+const TRANSACTION_VERSION: felt252 = 1;
+
+
+#[starknet::interface]
+trait IAccount<TContractState>{
+    fn __execute__(ref self: TContractState, calls:Array<Call>, token_contract:ContractAddress, token_id:u256) -> Array<Span<felt252>>;
+    fn __validate__(ref self: TContractState, calls:Array<Call>) -> felt252;
+    fn __validate_declare__(self:@TContractState, class_hash:felt252) -> felt252;
+    fn __validate_deploy__(self: @TContractState, class_hash:felt252, contract_address_salt:felt252, public_key:felt252) -> felt252;
+    fn set_public_key(ref self: TContractState, new_public_key:felt252);
+    fn get_public_key(self: @TContractState) -> felt252;
+    fn is_valid_signature(self:@TContractState, hash:felt252, signature:Array<felt252>) -> felt252;
+    // fn support_interface(self:@TContractState, interface_id:felt252) -> bool;
+    fn token(self:@TContractState) -> (ContractAddress, u256);
+    fn owner(ref self: TContractState, token_contract:ContractAddress, token_id:u256) -> ContractAddress; 
+    fn nonce(self:@TContractState) -> felt252;
+}
+
+#[starknet::interface]
+trait IERC721<TContractState> {
+    fn balance_of(self: @TContractState, account: ContractAddress) -> u256;
+    fn owner_of(self: @TContractState, token_id: u256) -> ContractAddress;
+    fn transfer_from(ref self: TContractState, from: ContractAddress, to: ContractAddress, token_id: u256);
+    fn safe_transfer_from(
+        ref self: TContractState, from: ContractAddress, to: ContractAddress, token_id: u256, data: Span<felt252>
+    );
+    fn approve(ref self: TContractState, to: ContractAddress, token_id: u256);
+    fn set_approval_for_all(ref self: TContractState, operator: ContractAddress, approved: bool);
+    fn get_approved(self: @TContractState, token_id: u256) -> ContractAddress;
+    fn is_approved_for_all(self: @TContractState, owner: ContractAddress, operator: ContractAddress) -> bool;
+    // IERC721Metadata
+    fn name(self: @TContractState) -> felt252;
+    fn symbol(self: @TContractState) -> felt252;
+    fn token_uri(self: @TContractState, token_id: u256) -> felt252;
+}
+
+#[starknet::contract]
+mod Account {
+
+use starknet::{get_tx_info, get_caller_address, get_contract_address};
+use ecdsa::check_ecdsa_signature;
+use array::SpanTrait;
+use array::ArrayTrait;
+use box::BoxTrait;
+use option::OptionTrait;
+use zeroable::Zeroable;
+use starknet::account::Call;
+use starknet::ContractAddress;
+use super::IERC721DispatcherTrait;
+use super::IERC721Dispatcher;
+
+    #[storage]
+    struct Storage{
+        _public_key: felt252,
+        _token_contract:ContractAddress,
+        _token_id:u256,
+    }
+
+
+
+        #[constructor]
+        fn constructor(ref self: ContractState, _public_key:felt252, token_contract:ContractAddress, token_id:u256){
+        self.initializer(_public_key, token_contract, token_id);
```

Comment:
I think delegating this to a helper function is a bit of an overkill. I prefer to keep codes as simple as possible, and considering this initializer function does not really perform any complex logic/computation aside writing to storage, I suggest we pull the whole logic directly into the constructor.